c++/src/capnp/compat/json.c++

```diff
@@ -386,7 +389,7 @@ void JsonCodec::decodeObject(JsonValue::Reader input, StructSchema type, Orphana
     KJ_IF_MAYBE(fieldSchema, type.findFieldByName(field.getName())) {
       decodeField(*fieldSchema, field.getValue(), orphanage, output);
     } else {
-      // Unknown json fields are ignored to allow schema evolution
+      KJ_REQUIRE(impl->ignoreUnrecognizedFields, "Unrecognized field.", field.getName());
```

Comment:
This is super-nitpicky but I'd remove the `.` from this string. It's not a complete sentence anyway, and KJ error output is going to add a `;` after it which would look weird.

Reply:
Guilty as charged, done.